src/org/zaproxy/zap/extension/formhandler/PopupMenuAddFormhandlerParam.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.formhandler;
+
+import java.awt.Component;
+
+import javax.swing.JTable;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.ExtensionPopupMenuItem;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.params.HtmlParameterStats;
+import org.zaproxy.zap.extension.params.ParamsPanel;
+import org.zaproxy.zap.extension.params.ParamsTableModel;
+
+public class PopupMenuAddFormhandlerParam extends ExtensionPopupMenuItem {
+
+	private static final long serialVersionUID = 1L;
+
+	private JTable paramTable = null;
+	private PopupDialogAddField popupDialogAddField = null;
+
+	public PopupMenuAddFormhandlerParam() {
+		super(Constant.messages.getString("formhandler.popup.menu.params.add.label"));
+		this.addActionListener(new java.awt.event.ActionListener() {
+
+			@Override
+			public void actionPerformed(java.awt.event.ActionEvent e) {
+				HtmlParameterStats hps = ((ParamsTableModel) paramTable.getModel()).getHtmlParameterStatsAtRow(paramTable.getSelectedRow());
+				popupDialogAddField = new PopupDialogAddField(View.getSingleton().getOptionsDialog(null), hps.getName(), hps.getValuesSummary().split(","));
+				popupDialogAddField.setVisible(true);
+			}
+		});
+	}
+
+	@Override
+	public boolean isEnableForComponent(Component invoker) {
+		if (invoker.getName() != null && invoker.getName().equals(ParamsPanel.PANEL_NAME)) {
+			this.paramTable = (JTable) invoker; 
+			if (paramTable.getSelectedRowCount() != 1) {
+				this.setEnabled(false);
+				return true;
+			}
+
+			HtmlParameterStats item = ((ParamsTableModel) paramTable.getModel()).getHtmlParameterStatsAtRow(paramTable.getSelectedRow());//(HtmlParameterStats) paramTable.getValueAt(paramTable.getSelectedRow(), 0);

```

Comment:
This additional check is redundant (unless you plan to do additional checks? Like if the type is URL or form), if we have a row selected we have an item. It could just do:
```Java
this.paramTable = (JTable) invoker; 
this.setEnabled(paramTable.getSelectedRowCount() == 1);
return true;
```

Reply:
I thought about checking type, but then decided not to. I guess I could add a note label to the dialog that makes it clear the form handler currently only deals with form parameters (actually there might be an existing message I can use).

Allowing headers might not be to useful but allowing cookies might be handy for certain HPP issues etc

Reply:
Agreed, better if not restricted to some types.